build/configure/build_config.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+var utils = require('utils');
+
+var BuildConfig = function(outputFilePath, defaultConfigFilePath) {
+  this.defaultConfigFilePath = defaultConfigFilePath;
+  this.outputFilePath = outputFilePath;
+  this.outpufConfig = {};
+  this.outputConfigContent = '';
+  this.start();
+};
+
+BuildConfig.prototype = {
+  DEFAULT_CONFIG_FILENAME: 'build_config.in',
+  DEFAULT_OUTPUT_CONFIG_FILENAME: 'build_config.in',
+
+  start: function() {
+    var defaultConfigFile = utils.getFile(
+      this.defaultConfigFilePath || this.DEFAULT_CONFIG_FILENAME);
+    if (defaultConfigFile.exists()) {
+      this.outputConfigContent +=
+        (utils.getFileContent(defaultConfigFile) + '\n');
+    }
+  },
+
+  get path() {
+    return this.outputFilePath || this.DEFAULT_OUTPUT_CONFIG_FILENAME;
+  },
+
+  insertConfig: function(name, value) {
+    // Prevent dup line
+    this.outpufConfig[name] = value;
+  },
+
+  writeConfig: function() {
```

Comment:
While you're doing templating thing, it may be better if you could give an example of input and output in the comment.